src/Server/Sanitizer.fs

```diff
@@ -13,59 +13,186 @@ let unsafe = ["System"; "NativePtr"]
+    ident |> isLongIdentIncludeUnsafe
+  | _ -> false
+
 let rec visitExpr = function
+  // Application
   | SynExpr.App (_, _, f, arg, _) ->
     visitExpr f
     || visitExpr arg
+
+  // Long Identifier
   | SynExpr.LongIdent (_, LongIdentWithDots (ident, _), _, _) ->
     ident |> isLongIdentIncludeUnsafe
-  | SynExpr.Match (_, e, clauses, _) ->
+
+  // Match expressions
+  | SynExpr.Match (_, e, clauses, _)
+  | SynExpr.MatchBang (_, e, clauses, _) ->
     visitExpr e
-    || clauses |> List.fold (fun acc (Clause (_, wh, e, _, _)) ->
-      acc
-      || visitExpr e
-      || (match wh with | Some e -> visitExpr e | None -> false)) false
+    || clauses |> List.fold visitClause false
+  | SynExpr.MatchLambda (_, _, clauses, _, _) ->
+    clauses |> List.fold visitClause false // Maybe List.exists?
+
+  // If-Then-Else expression
   | SynExpr.IfThenElse (g, t, f, _, _, _, _) ->
     visitExpr g
     || visitExpr t
     || (match f with | Some f -> visitExpr f | None -> false)
+
+  // Let/Use expression
   | SynExpr.LetOrUse (_, _, bindings, body, _) ->
     bindings
-    |> List.forall (fun (Binding (_, _, _, _, _, _, _, _, _, e, _, _)) ->
+    |> List.forall (fun (Binding (_, _, _, _, _, _, _, _, _, e, _, _)) -> // Why is this forall?
       visitExpr e)
     || visitExpr body
   | SynExpr.LetOrUseBang (_, _, _, _, rhs, _, body, _) ->
     visitExpr rhs || visitExpr body
+
+  // Array/List expression
   | SynExpr.ArrayOrList (_, exprs, _) ->
     exprs |> List.exists visitExpr
   | SynExpr.ArrayOrListOfSeqExpr (_, e, _) ->
     visitExpr e
-  | SynExpr.CompExpr (_, _, e, _) ->
-    visitExpr e
+
+  // Loop expressions
   | SynExpr.ForEach (_, _, _, _, enum, body, _) ->
-    visitExpr enum
-    || visitExpr body
+    visitExpr enum || visitExpr body
+  | SynExpr.For (_, _, f, _, t, d, _) ->
+    visitExpr f || visitExpr t || visitExpr d
+  | SynExpr.While (_, cond, body, _) ->
+    visitExpr cond || visitExpr body
+
+  // Sequential, Tuple expression
   | SynExpr.Sequential (_, _, e1, e2, _) ->
-    visitExpr e1
-    || visitExpr e2
+    visitExpr e1 || visitExpr e2
+  | SynExpr.Tuple (_, exprs, _, _) ->
+    exprs |> List.exists visitExpr
+
+  // Exception expressions
+  | SynExpr.TryFinally(t, f, _, _, _) ->
+    visitExpr t || visitExpr f
+  | SynExpr.TryWith(body, _, cases, _, _, _, _) ->
+    visitExpr body
+    || cases |> List.fold visitClause false
+
+  // Get/Set expressions
+  | SynExpr.Set (id, e, _) ->
+    visitExpr id || visitExpr e
+  | SynExpr.LongIdentSet (LongIdentWithDots (ident, _), e, _) ->
+    ident |> isLongIdentIncludeUnsafe
+    || visitExpr e
+  | SynExpr.NamedIndexedPropertySet (LongIdentWithDots (ident, _), i, e, _) ->
+    ident |> isLongIdentIncludeUnsafe
+    || visitExpr i || visitExpr e
+  | SynExpr.DotGet (e, _, _, _) ->
+    visitExpr e
+  | SynExpr.DotIndexedGet (e, indices, _, _) ->
+    visitExpr e
+    || indices |> List.exists visitIndex
```

Comment:
Seems like indices are the only one that  needs checking.

Reply:
One can find an offending dot-indexable expression, for example `System.Diagnostics.Process.Start("cmd.exe","").Threads.[0]`

Reply:
Very good.